### PR TITLE
Gendocs for docs/admin/kube-*

### PR DESCRIPTION
**Pull request by feiskyer**
CC @thockin

### CODE REVIEW

**Comment by bgrant0607** [comment]
cc @mikedanese

**Comment by feiskyer** [comment]
Rebased. @bgrant0607 @thockin @eparis

**Comment by feiskyer** [comment]
@dchen1107 @yujuhong @mikedanese I think this PR is reasonable for v1.1 to update docs for `admin/kube-*`

**Comment by thockin** [comment]
This looks pretty good to me.  My biggest concern in the number of places that each binary has to be spelled out.  I wish there were a way to mitigate that.  Any ideas?

@mikedanese

**Comment by mikedanese** [comment]
@thockin we need https://github.com/kubernetes/kubernetes/issues/14541

can we add this to update-generated-docs.sh and add it to precommit/ci checks as well?

File: cmd/genkubedocs/gen_kube_docs.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+)
+
+func main() {
+	// use os.Args instead of "flags" because "flags" will mess up the man pages!
+	path := "docs/"
+	module := "kubelet"
+	if len(os.Args) == 3 {
+		path = os.Args[1]
+		module = os.Args[2]
+	} else if len(os.Args) > 3 {
```
Comment (mikedanese) [inline cmd/genkubedocs/gen_kube_docs.go]: We should not have any defaults and assert that we have len(os.Args) == 3 (i.e. make the > a !=)

**Comment by eparis** [comment]
https://github.com/kubernetes/kubernetes/pull/13614 is an alternative approach, just use cobra and do away with all of these extra binaries.  I really should rebase it....

File: cmd/genkubedocs/gen_kube_docs.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	outDir, err := genutils.OutDir(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get output directory: %v\n", err)
+		os.Exit(1)
+	}
+
+	switch module {
+	case "apiserver":
```
Comment (mikedanese) [inline cmd/genkubedocs/gen_kube_docs.go]: can we use the full names? "kube-apiserver"

**Comment by mikedanese** [comment]
My concern would be, this approach does not seem robust to global flag cross polination. I can't see any examples in the output though.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

File: cmd/genkubedocs/gen_kube_docs.go (hunk 1 above)
Comment (feiskyer) [inline cmd/genkubedocs/gen_kube_docs.go]: OK, I'll modify.

File: cmd/genkubedocs/gen_kube_docs.go (hunk 2 above)
Comment (feiskyer) [inline cmd/genkubedocs/gen_kube_docs.go]: OK

**Comment by feiskyer** [comment]
@mikedanese Rebased. Don't know why "No test results found." for GCE e2e test.

**Comment by jlowdermilk** [comment]
Sorry about failing tests. I broke e2e job trying to add in unit/integration. Should be working now. @k8s-bot test this

**Pull request merged**